### ISSUE TITLE
Accessibility: Failed Listview Accessibility Tests

### ISSUE THREAD

**Issue by clepore** [infor-design/enterprise#403]
Original Issue: http://jira.infor.com/browse/SOHO-7938

----

Axe is reporting the following WCAG2AA errors as critical errors on a handful of Listview component test pages:

- http://localhost:4000/components/listview/example-index:

> _Description:_
> Ensures ARIA attributes are allowed for an element's role 
> 
> _Fix any of the following:_
> ARIA attribute is not allowed: aria-selected="true"...
>```
> <li class="selected" aria-selected="true" tabindex="0">
>  <a href="#" aria-disabled="true"><span class="audible">You are currently on page  </span>1</a>
> </li>
>```
> Breaks axe rule: `aria-allowed-attr`

- http://localhost:4000/components/listview/example-mixed-selection
- http://localhost:4000/components/listview/remove-clear

![Screen Shot 2018-05-11 at 4 35 50 PM](https://user-images.githubusercontent.com/742319/55835147-692dcd80-5ae9-11e9-8582-81716edc692f.png)

> _Description:_
> Ensures every form element has a label
>
> _Help:_
> Form elements must have labels for about 12 nodes...
>
> _Fix any of the following:_
> - `aria-label` attribute does not exist or is empty
> - `aria-labelledby` attribute does not exist, references elements that do not exist or references elements that are empty or not visible
> - Form element does not have an implicit (wrapped) `<label>`
> - Form element does not have an explicit `<label>`
> - Element has no title attribute or the title attribute is empty
>```
> <input tabindex="-1" type="checkbox" class="checkbox hide-focus">
>```
> _Breaks axe rules:_
>   - `aria-label`
>   - `aria-labelledby`
>   - `implicit-label` 
>   - `explicit-label` 
>   - `non-empty-title`

- http://localhost:4000/components/listview/example-mixed-selection:

> _Description:_
> Ensures ARIA attributes are allowed for an element's role
>
> _Help:_
> Elements must only use allowed ARIA attributes
> 
> _Fix any of the following:_
>  - ARIA attribute is not allowed: `aria-selected="true"`
>```
> <li class="selected" aria-selected="true" tabindex="0">
>   <a href="#" aria-disabled="true"><span class="audible">You are currently on page  </span>1</a>
> </li>
>```
> _Breaks axe rules:_
>  - `aria-allowed-attr`

Follow debugging instructions on docs/TESTING.md to investigate the issue in the devTools console, and run test in high-contrast which has disables zero rules. These errors can also be double checked with Chrome Audit in devTools, or use the Chrome or Firefox extension, https://www.axe-core.org/

**Comment by tmcconechy** [comment]
Also as per #1721 - the listbox incorrectly has a `role="presentation"` not sure if thats flagged but noting it here

**Comment by EdwardCoyle** [comment]
Unassigned myself for now, just in case we don't get to it this sprint.  If anyone wants to pick this up and needs some details, feel free to ping me.

**Comment by EdwardCoyle** [comment]
Leaving this on hold for the moment.  After looking at the elements that were being reported, I realized that many of them are "implicit" label/inputs (read: the label wraps the input field, so there's no need for other `aria` labels or matching `for`/`id` attributes).  See my video of dev tools against Axe's reporting below:

![Apr-18-2019-4_00-PM](https://user-images.githubusercontent.com/3249601/56389064-9881c000-61f6-11e9-9b3a-2060ecfbe518.gif)

I'm going to look through their [issue tracker](https://github.com/dequelabs/axe-core/issues) to see there are any reported issues related to this, or maybe open one.

**Comment by EdwardCoyle** [comment]
Pushed a branch [`403-listview-accesibility`](https://github.com/infor-design/enterprise/tree/403-listview-accesibility) that has my terminal output tool, and enables the broken listview e2e tests so someone else can pick this up and test while I'm out, if we need to.

**Comment by janahintal** [comment]
QA Passed. Will move to done. Thanks
Tested in: 
http://master-enterprise.demo.design.infor.com/components/listview/example-index
http://master-enterprise.demo.design.infor.com/components/listview/example-mixed-selection
http://master-enterprise.demo.design.infor.com/components/listview/remove-clear
